src/main/scala/com/example/spark/app/SampleSparkApp.scala

```diff
@@ -1,35 +1,47 @@
 package com.example.spark.app
 
+import com.example.spark.app.Models.FootballTeam
 import org.apache.log4j.{Level, Logger}
-import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
 import org.apache.spark.sql.functions._
 
 
 /**
-  * This is an example Spark Application.
-  *
-  * To run this locally use `mvn test`
-  * or change Scala and Spark dependencies from `provided` to `compile`
-  *
-  * To run this on cluster generate the uber jar by running
-  * `mvn package` and use `spark-submit` on cluster.
-  */
+ * This is an example Spark Application.
+ *
+ * To run this locally use `mvn test`
+ * or change Scala and Spark dependencies from `provided` to `compile`
+ *
+ * To run this on cluster generate the uber jar by running
+ * `mvn package` and use `spark-submit` on cluster.
+ */
+
+
 object SampleSparkApp {
 
+  def generate_dataset(): Unit = {
+    val player = FootballTeam("artem", "league")
+  }
+
   def main(args: Array[String]): Unit = {
 
     Logger.getLogger("org").setLevel(Level.WARN)
     Logger.getLogger("akka").setLevel(Level.WARN)
 
+    generate_dataset()
+
     val spark = SparkSession
       .builder()
       .appName("Example SparkApplication")
       .master("local[*]")
       .getOrCreate()
 
+
+    // TODO Is that ok? importing from instantiated object
```

Comment:
Yep. It's a tricky bit that allows some extension methods (Spark magic) possible. Like invoking `.toDF` on a Seq.

In IntelliJ Click on menu 'View/Show Implicit Hints' to see what's going on beneath the surface of the code.